crates/wash-lib/src/parser/mod.rs

```diff
@@ -510,3 +544,63 @@ impl RawProjectConfig {
         })
     }
 }
+
+impl ProjectConfig {
+    pub fn resolve_registry_url(&self) -> Result<String> {
+        let registry_url = match &self.common.registry.url {
+            Some(url) => url.clone(),
+            None => {
+                bail!("No registry URL specified in wasmcloud.toml");
+            }
+        };
+
+        Ok(registry_url)
+    }
+
+    pub async fn resolve_registry_credentials(
+        &self,
+        registry: impl AsRef<str>,
```

Comment:
Should this be `registry`, or `artifact_url`? I noticed all the uses of this function pass a full artifact URL, not just the name of the registry